### PR TITLE
Add `code.stacktrace` attribute

### PR DESCRIPTION
**Pull request by AlexanderWert**
## Changes

There are use cases when it's desirable to collect a stacktrace that is not necessarily attached to an exception (e.g. to identify the exact stack of a span).

This PR proposes a new attribute `code.stacktrace` for the above-mentioned  use cases.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions. (N/A)

### CODE REVIEW

File: docs/general/attributes.md [hunk 1]
```diff
@@ -390,6 +390,7 @@ about the span.
 | `code.function` | string | The method or function name, or equivalent (usually rightmost part of the code unit's name). | `serveRequest` | Recommended |
 | `code.lineno` | int | The line number in `code.filepath` best representing the operation. It SHOULD point within the code unit named in `code.function`. | `42` | Recommended |
 | `code.namespace` | string | The "namespace" within which `code.function` is defined. Usually the qualified class or module name, such that `code.namespace` + some separator + `code.function` form a unique identifier for the code unit. | `com.example.MyHttpService` | Recommended |
+| `code.stacktrace` | string | A stacktrace as a string in the natural representation for the language runtime. The representation is to be determined and documented by each language SIG. | `at com.example.GenerateTrace.methodB(GenerateTrace.java:13)\n at com.example.GenerateTrace.methodA(GenerateTrace.java:9)\n at com.example.GenerateTrace.main(GenerateTrace.java:5)` | Recommended |
```
Comment (KarstenSchnitter) [inline docs/general/attributes.md:393]: Following https://github.com/open-telemetry/opentelemetry-specification/blob/v1.26.0/specification/common/attribute-requirement-level.md I suggest to have the stacktrace as "Opt-In". The commit message already states, that there are only "certain use-cases" that would benefit. Stacktraces usually increase the event size quite substantially and come with a performance cost. Exceptions should occur in a low frequency and greatly benefit from the additional information. Normal events usually do not. Therefore, I feel "Opt-In" would be a more suitable approach.

Comment (AlexanderWert) [reply docs/general/attributes.md:393]: Absolutely agree!

We would move all of the `code` attributes into [the registry](https://github.com/open-telemetry/semantic-conventions/tree/main/model/registry) anyways. And in the context of concrete usages of that field (e.g. in the general attributes semantic conventions we could overwrite it with req-level `Opt-In`)

See this PR: https://github.com/open-telemetry/semantic-conventions/pull/434

**Review by trask — COMMENTED** [review]
Would we deprecate `exception.stacktrace` in favor of this generalization?

**Comment by AlexanderWert** [comment]
> Would we deprecate `exception.stacktrace` in favor of this generalization?

@trask I wasn't intending to imply changes to `exception.stacktrace` with that, but I'm open to it if we want to. `code.stacktrace` would cover other non-exception-related use cases. Also as you can see in the example value in this case the exception message is not present here.

But, you are right that `code.stacktrace` is a more general attribute that theoretically could be reused / referenced in the exception semantic conventions.

**Comment by pyohannes** [comment]
> Would we deprecate exception.stacktrace in favor of this generalization?

Re-using a generic `stacktrace` field might be another option (hypothetical until there's progress on #339).

**Comment by trask** [comment]
> Also as you can see in the example value in this case the exception message is not present here.

👍

**Review by jack-berg — APPROVED** [review]
I have also had use cases for including a stack trace outside of an exception and came to the same conclusion that the attribute should be named `code.stacktrace`.

Thanks for beating me too it 😁!

File: docs/general/attributes.md (hunk 1 above)
Comment (trask) [inline docs/general/attributes.md:393]: Is there an equivalent to these `code.*` attributes in ECS?

Comment (AlexanderWert) [reply docs/general/attributes.md:393]: no, we don't have something like that in ECS defined

**Review by trask — APPROVED** [review]

**Comment by Oberon00** [comment]
I didn't read everything, but please note that exception.stacktrace is not a plain stacktrace but the default detailed textual exception representation (including, usually, exception type, message, caused-by, etc)

**Comment by AlexanderWert** [comment]
> I didn't read everything, but please note that exception.stacktrace is not a plain stacktrace but the default detailed textual exception representation (including, usually, exception type, message, caused-by, etc). There was a long discussion about the name (IIRC I was strongly against exceptions.stacktrace since in many languages there is also the possibility to get a textual representation of just the exception stacktrace), but there is no other common name for that concept.

Thanks @Oberon00 , that's also how I understood `exception.stacktrace`. For `code.stacktrace` the intention is to capture the textual representation of just the exception stacktrace. What do you think in general about the proposal in this PR?

**Comment by trask** [comment]
should this have a change log entry?

**Review by pyohannes — APPROVED** [review]

**Pull request merged**